src/main/java/com/google/devtools/build/lib/skyframe/PrerequisiteProducer.java

```diff
@@ -405,6 +429,30 @@ public boolean evaluate(
     return true;
   }
 
+  private PlatformInfo loadTargetPlatformInfo(Environment env)
+      throws InterruptedException, ToolchainException {
+    if (!targetAndConfiguration.getConfiguration().hasFragment(PlatformConfiguration.class)) {
+      return null;
+    }
+    // Load target platform info through toolchain resolution with no toolchain types to prevent a
+    // direct Skyframe dependency on the target platform - this would lead to incorrect cquery
+    // output.
+    ToolchainContextKey toolchainContextKey = ToolchainContextKey.key()
```

Comment:
What was the previous version, just directly creating a ConfiguredTargetKey for the target platform? This feels like a lot of overhead, I'd rather do the simple thing here and then fix cquery, rather than this hack.

Reply:
@gregestren to discuss fixing cquery

Reply:
The previous version used `PlatformLookupUtil#getPlatformInfo` and directly constructed the `ConfiguredTargetKey` for the target platform from the `BuildConfiguration`. The current approach is of similar complexity just going by lines of code.

Reply:
Yes, but creating the CTK is more direct, this is causing unneeded toolchain resolution just to avoid a dependency edge.

Reply:
Yes, updating the test makes sense. Thanks very much for these changes!